### PR TITLE
Workflow improvements.

### PR DESCRIPTION
**Pull request by tikurahul**
* Makes adding more modules easier.
* Remove the `pull_request` hook in favor of `pull_request_target`.
* Add secrets back to the hooks.

Test: Workflow tests.

### CODE REVIEW

**Comment by tikurahul** [comment]
Also: https://github.com/tikurahul/androidx/runs/947050349

File: .github/workflows/presubmit.yml [hunk 1]
```diff
@@ -72,26 +71,23 @@ jobs:
...
       - name: Upload Results
         if: always()
         uses: actions/upload-artifact@v2
         with:
           name: artifacts
           path: ~/dist
-      - name: Completed WebHook
-        if: always()
+      - name: Successful WebHook
+        if: ${{ success() }}
```
Comment (dlam) [inline .github/workflows/presubmit.yml]: Is the ${{ }} necessary here?

Comment (dlam) [reply .github/workflows/presubmit.yml]: i.e., why isn't it needed for `always()`

File: .github/workflows/presubmit.yml [hunk 2]
```diff
@@ -72,26 +71,23 @@ jobs:
...
+        uses: tikurahul/github-workflow-webhook-action@master
+        with:
+          url: 'https://androidx.dev/github/androidX/presubmit/hook'
+          secret: ${{ secrets.ANDROIDX_PRESUBMIT_HOOK_SECRET }}
+          payload: '{ "platform": "${{ matrix.os }}", "token": "${{ secrets.GITHUB_TOKEN }}", "state": "completed", "success": true }'
```
Comment (dlam) [inline .github/workflows/presubmit.yml:86]: it kinda sucks that we can't inline the if failure / success check here? That looks like the only difference to me from these two branches

File: .github/workflows/presubmit.yml [hunk 3]
```diff
@@ -72,26 +71,23 @@ jobs:
...
+      - name: Failure WebHook
+        if: ${{ failure() || cancelled() }}
```
Comment (dlam) [inline .github/workflows/presubmit.yml:88]: Are these states based on previous step or the entire job? e.g., what happens if:
```
- name: bOS fails
  ...
- name: always succeeds
  if: always()
  run: true
- name Successful WebHook
  if: success()
  run: echo "this shouldn't run!"
```

**Review by dlam — APPROVED** [review]

File: .github/workflows/presubmit.yml (hunk 2 above)
Comment (tikurahul) [reply .github/workflows/presubmit.yml:86]: Yes, it does. But unfortunately there is no easy way to inline this without having to write more actions. :disappointed:

File: .github/workflows/presubmit.yml (hunk 3 above)
Comment (tikurahul) [reply .github/workflows/presubmit.yml:88]: These are based on the entire job: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions has more information.

File: .github/workflows/presubmit.yml (hunk 1 above)
Comment (tikurahul) [reply .github/workflows/presubmit.yml]: I just copied the example at https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions. 

I think this is something I can probably get rid of.

Comment (tikurahul) [reply .github/workflows/presubmit.yml]: Done.

**Pull request closed without merging**